on_device_training/mobile/ios/MyVoice/TrainView.swift

```diff
@@ -0,0 +1,146 @@
+        "With every page turned in a book, I open the door to new worlds and ideas. The written words carry the wisdom of countless souls, and I am humbled by the knowledge they offer. In stories, I find a mirror to my own experiences and a beacon of hope for a better tomorrow.",
+        "Life's trials may bend me, but they will not break me. Through adversity, I discover the strength within my heart. Each obstacle is a chance to learn, to evolve, and to emerge as a better version of myself. I am grateful for every lesson, for they shape me into the person I am meant to be.",
+        "The sky above is an ever-changing canvas of colors and clouds. In its vastness, I realize how small I am in the grand scheme of things, and yet, I know my actions can ripple through the universe. As I walk this Earth, I seek to leave behind a positive impact and a legacy of love and compassion.",
+        "In the stillness of meditation, I connect with the depth of my soul. The external noise fades away, and I hear the whispers of my inner wisdom. With each breath, I release tension and embrace serenity. Meditation is my sanctuary, a place where I can find clarity and renewed energy.",
+        "Kindness is a chain reaction that spreads like wildfire. A simple act of compassion can brighten someone's day and inspire them to pay it forward. Together, we can create a wave of goodness that knows no boundaries, reaching even the farthest corners of the world.",
+        "As the sun rises on a new day, I am filled with gratitude for the gift of life. Every moment is a chance to make a difference, to love deeply, and to embrace joy. I welcome the adventures that await me and eagerly embrace the mysteries yet to be uncovered."
+    ]
+
+    
+    private let knumRecording = 5
+    private let audioRecorder = AudioRecorder()
+    private let trainer = try! Trainer(sampleAudioRecordings: sentences.count)
+    
+    @State private var trainingData: [Data] = []
+    @State private var currentSentenceIndex = 0
+    @State private var readyToRecord: Bool = true
+    @State private var isRecordingComplete = false
+    @State private var isTrainingInProgress = false
+    
+    private func recordVoice() {
+        audioRecorder.record { recordResult in
+           switch recordResult {
+           case .success(let recordingData):
+               trainingData.append(recordingData)
+               print("Successfully completed Recording")
+           case .failure(let error):
+               print("Error: \(error)")
+            }
+            
+            readyToRecord = true
+            
+            if currentSentenceIndex < knumRecording - 1 {
+                currentSentenceIndex += 1
```

Comment:
do we need a separate `currentSentenceIndex` variable or could we use `trainingData.count`?

also, maybe we could use `trainingData.count % sentences.count` to determine which sentence to display? then it doesn't rely on `knumRecording <= sentences.count`.